task/ab-load/0.1/ab-load.yaml

```diff
@@ -0,0 +1,161 @@
+          }
+
+          # execute the AB benchmark for GET load
+          function abGET {
+            abDefault GET "$url" "$size" "$concurrency" "$timeout" "$additionnal"
+          }
+
+          # execute the AB benchmark for POST load
+          function abPOST {
+            local file=${AB_FILE}
+            local type=${AB_TYPE:-application/x-www-form-urlencoded}
+            local url=${1:-myurl}
+            local size=${2:-1000}
+            local concurrency=${3:-10}
+            local timeout=${4:-10}
+            local additionnal=${5}
+            echo "= Start http test POST ${AB_URL}"
+            echo "  - ${size} requests"
+            echo "  - ${concurrency} concurrent jobs"
+            ab -n "$size" \
+               -c "$concurrency" \
+               -s "$timeout" \
+               -p "$file" \
+               -T '${type}; charset=UTF-8' \
+            $additionnal "$url"
+          }
+
+          # execute the AB benchmark for PUT load
+          function abPUT {
+            local file=${AB_FILE}
+            local type=${AB_TYPE:-application/x-www-form-urlencoded}
+            local url=${1:-myurl}
+            local size=${2:-1000}
+            local concurrency=${3:-10}
+            local timeout=${4:-10}
+            local additionnal=${5}
+            echo "= Start http test PUT ${AB_URL}"
+            echo "  - ${size} requests"
+            echo "  - ${concurrency} concurrent jobs"
+            ab -n "$size" \
+               -c "$concurrency" \
+               -s "$timeout" \
+               -u "$file" \
+               -T '${type}; charset=UTF-8' \
+            $additionnal "$url"
+          }
+
+          # execute the AB benchmark for DELETE load
+          function abDELETE {
+            abDefault DELETE "$url" "$size" "$concurrency" "$timeout" "$additionnal"
+          }
+
+          if [[ $AB_METHOD == 'DELETE' ]]
+          then
+              abDELETE "$AB_URL" "$AB_SIZE" "$AB_CONCURRENCY" "$AB_TIMEOUT" "$AB_ADDITIONNAL_FLAGS"
+          elif [[ $AB_METHOD == 'POST' ]]
+          then
+              abPOST "$AB_URL" "$AB_SIZE" "$AB_CONCURRENCY" "$AB_TIMEOUT" "$AB_ADDITIONNAL_FLAGS"
+          elif [[ $AB_METHOD == 'PUT' ]]
+          then
+              abPUT "$AB_URL" "$AB_SIZE" "$AB_CONCURRENCY" "$AB_TIMEOUT" "$AB_ADDITIONNAL_FLAGS"
+          else
+              abGET "$AB_URL" "$AB_SIZE" "$AB_CONCURRENCY" "$AB_TIMEOUT" "$AB_ADDITIONNAL_FLAGS"
+          fi
+      env:
+        - name: AB_URL
+          value: $(params.url)
+        - name: AB_SIZE
+          value: $(params.size)
+        - name: AB_CONCURRENCY
+          value: $(params.concurrency)
+        - name: AB_TIMEOUT
+          value: $(params.timeout)
+        - name: AB_METHOD
+          value: $(params.method)
+        - name: AB_ADDITIONNAL_FLAGS
+          value: $(params.additionnalFlags)
+        - name: AB_FILE
+          value: /tmp/postfile/content
```

Comment:
can we make this more flexible, using the filename as `content` will bind the user to create the configmap with this name OR can we add a section in readme stating the requirements of creating the configmap so that the users are aware of the hard requirements?

Reply:
changed with only the filename to use (coresponding to a key in cm, secret or pvc)